presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1584,7 +1584,7 @@ public PhysicalOperation visitIndexJoin(IndexJoinNode node, LocalExecutionPlanCo
             }
 
             OperatorFactory lookupJoinOperatorFactory;
-            OptionalInt totalOperatorsCount = getJoinOperatorsCountForSpill(context, session);
+            OptionalInt totalOperatorsCount = context.getDriverInstanceCount();
```

Comment:
since Lookup Join does not support spill, would be good not to pass `partitioningSpillerFactory` as well.
Eg.

```
// There is no spill in lookup join's case
OptionalInt totalOperatorsCount = context.getDriverInstanceCount();
PartitioningSpillerFactory partitioningSpillerFactory = unsupportedPartitioningSpillerFactory();
```

Reply:
Also, i am not sure about cmt msg "Fix index join obtaining driver instance count when spill is enabled". Can you rephrase it? From what i can understand, you're fixing general bug, as query with index join right above table scan and with spill enabled could fail. If that is correct then this would be more appropriate:
```
Fix query failure when spill enabled and there is Index Join above Table Scan
```